[5.09.30] Bricked Blast Furnace

JohannesGaessler opened the pull request:
As a way of rebalancing the Bronze Age I have added a Blast Furnace that is built from bricks.
This eliminates the need to gather large amounts of Bronze for a Bronze Blast Furnace.

![2017-06-08_22 51 52](https://user-images.githubusercontent.com/18492268/27256664-47afcbe8-53bd-11e7-9c59-e78d9defc0c0.png)


The Bricked Blast Furnace has the same structure as the Bronze Blast Furnace and is built from Firebricks.
Firebricks are made from clay, but they require more complex processing than vanilla Bricks.

To build a Bricked Blast Furnace the player needs to build a Steam Compressor and a Steam Hammer/Steam Macerator.
The first Bricked Blast Furnace requires a total of 130 Clay blocks to build.
This is comparable to the 150 Bronze required for a Bronze Blast Furnace, but clay is found much more common than Copper and Tin, therefore reducing the amount of time the player needs to spend on searching for a vein.

On another note, NEI recipes for the primitive blast furnaces have been added.

Commit:
Implemented Bricked Blast Furnace, added corresponding textures

Added a new Casing for the Bricked Blast Furnace, GT_Block_Casings4, ID
15

Refactored the non-electric Blast Furnaces:
The Bronze Blast Furnace and the Bricked Blast Furnace extend a
superclass to ensure consistent behavior.
Subclasses only differ in texture, description, and Casing Block.

Commit:
Added a Recipe Map for Primitive Blast Furnaces

The Recipe Map is not yet in use.

Commit:
Changed the Primitive Blast Furnace so that it uses the new Recipe Map

Commit:
Added Interface classes for the BBF, increased slots to 3/3

Commit:
Adjusted BBF GUIs to accommodate to the new 3/3 layout.

Commit:
Merge remote-tracking branch 'GT5-Unofficial-Fork/ChemistryUpdate' into
BrickedBlastFurnace

Conflicts:
	src/main/java/gregtech/api/enums/Materials.java
	src/main/java/gregtech/api/enums/Textures.java
	src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_BronzeBlastFurnace.java
	src/main/java/gregtech/loaders/postload/GT_MachineRecipeLoader.java
	src/main/java/gregtech/loaders/preload/GT_Loader_MetaTileEntities.java

Commit:
Implemented crafting recipes for the Bricked Blast Furnace

fixed Glass Hammering consuming 40x more energy than intended

Fixed Hammer recipes that (presumably unintentionally) had 10EUt, 16
ticks instead of 16EUt, 10 ticks.

Renamed Dense Bricks -> Firebricks

Rongmario commented:
There is a complete GT version up for compile with this and the OilRefinery update on my fork of GT5U.

draknyte1 commented on src/main/java/gregtech/common/GT_RecipeAdder.java:
```diff
@@ -143,6 +145,26 @@ public boolean addBlastRecipe(ItemStack aInput1, ItemStack aInput2, FluidStack a
+        }
+        ItemStack[] coalStacks = new ItemStack[]{Materials.Coal.getGems(aCoalAmount), Materials.Coal.getDust(aCoalAmount), Materials.Charcoal.getGems(aCoalAmount), Materials.Charcoal.getDust(aCoalAmount)};
+        for (ItemStack coalStack : coalStacks) {
+        	GT_Recipe.GT_Recipe_Map.sPrimitiveBlastRecipes.addRecipe(true, new ItemStack[]{aInput1, aInput2, coalStack}, new ItemStack[]{aOutput1, aOutput2, Materials.Ash.getDustTiny(aCoalAmount)}, null, null, null, null, aDuration, 0, 0);
```
Normal Coal and Coke produce different types of ash..

draknyte1 commented on src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PrimitiveBlastFurnace.java:
```diff
@@ -0,0 +1,326 @@
+	}
+
+	public MetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
+		return new GT_MetaTileEntity_BronzeBlastFurnace(this.mName);
```
Is this intended?

Techlone replied:
The copy-paste mistake
Happens

draknyte1 replied:
Just pointing things out during my review.

draknyte1 reviewed (changes requested):
So, 95% of it looks good.
Just be sure to change the things noted and add a config and I see no issues otherwise.

Also be sure the BBF uses the BBF GUI and TE, whilst the bricked one uses the Bricked GUI and TE.
Don't remember which the BBF uses, the diffing is ugh online due to everything having moved a line.

Commit:
Added Bricked BF config, fixed Primitive BF class hierarchy

Blast Furnace Recipes using Coal or Charcoal now yield Dark Ash instead
of Ash.

Techlone commented on src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_BrickedBlastFurnace.java:
```diff
@@ -29,7 +29,7 @@ public MetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
                 "Useable for Steel and general Pyrometallurgy",
                 "Size(WxHxD): 3x4x3 (Hollow, with opening on top)",
-                "Built from 32 Bricks",
+                "Built from 32 Firerick blocks",
```
*fire**b**rick
